mongo/src/main/java/com/redhat/lightblue/mongo/crud/BasicDocSaver.java

```diff
@@ -288,59 +293,35 @@ private void updateDocs(CRUDOperationContext ctx,
                         try {
                             ctx.getFactory().getInterceptors().callInterceptors(InterceptPoint.PRE_CRUD_UPDATE_DOC, ctx, doc.inputDoc);
                             merge.merge(doc.oldDoc, doc.newDoc);
-                            Translator.populateDocHiddenFields(doc.newDoc, md);
+                            Translator.populateCaseInsensitiveHiddenFields(doc.newDoc, md);
                             updateAttemptList.add(doc);
                         } catch (Exception e) {
                             doc.inputDoc.addError(Error.get("update", MongoCrudConstants.ERR_TRANSLATION_ERROR, e));
                         }
                     } else {
                         doc.inputDoc.addError(Error.get("update",
-                                CrudConstants.ERR_NO_FIELD_UPDATE_ACCESS, paths.toString()));
+                                                        CrudConstants.ERR_NO_FIELD_UPDATE_ACCESS, paths.toString()));
                     }
                 }
                 LOGGER.debug("After checks and merge, updating {} docs", updateAttemptList.size());
                 if (!updateAttemptList.isEmpty()) {
-                    BulkWriteOperation bw = collection.initializeUnorderedBulkOperation();
+                    BulkUpdateCtx updateCtx=new BulkUpdateCtx();
+                    updateCtx.reset(collection);
```

Comment:
I guess if you did that, it might also make sense to remove collection from the execute param. Is there a reason this is a param? I would think to correctly find any errors, it would have to be the same collection that was passed to reset, anyway? So maybe passing in constructor helps avoid a class of bugs if I understand correctly.